ethereum_history_api/circuits/lib/src/misc/bytes.nr

```diff
@@ -0,0 +1,53 @@
+use dep::proof::utils::byte_value;
+
+pub fn left_nibble_shift<N>(nibbles: [u8; N]) -> ([u8; N], u64) {
+    let (mut shifted, mut len) = byte_value(nibbles);
+    if (len % 2 == 1) {
+        right_shift_by_one(&mut shifted);
+        len += 1;
+    }
+
+    let shifted_by_len = N - len;
+    (shifted, shifted_by_len)
+}
+
+fn right_shift_by_one<N>(arr: &mut [u8; N]) {
+    let mut last = 0;
+    for i in 0..N {
+        let tmp = (*arr)[i];
+        (*arr)[i] = last;
+        last = tmp;
```

Comment:
Do you think that extracting a swap function will improve the readability here?

Reply:
I just implemented swap and it would need to take 2 arguments by reference*. But unfortunately Noir doesn't support mutable references to array elements so I wouldn't be able to use it here. So it doesn't make sense to me to implement swap

\* - If we don't pass arguments by reference, we'd need to create 2 new variables just to read the output of swap, which doesn't improve readability